atom/browser/api/atom_api_app.cc

```diff
@@ -714,6 +722,48 @@ void App::SetLocale(std::string locale) {
   g_browser_process->SetApplicationLocale(locale);
 }
 
+std::string App::GetCountryName() {
+  std::string country = "";
+
+  #if defined(OS_WIN)
+  // for more info, see:
+  // https://docs.microsoft.com/en-us/windows/desktop/api/winnls/nf-winnls-getlocaleinfoex
+  // https://docs.microsoft.com/en-us/windows/desktop/Intl/locale-slocalized-constants
+  WCHAR localized_country_name[80];
+  int country_name_length = GetLocaleInfoEx(
+    LOCALE_NAME_USER_DEFAULT,
+    LOCALE_SLOCALIZEDCOUNTRYNAME,
+    (LPWSTR)&localized_country_name,
+    sizeof(localized_country_name) / sizeof(WCHAR));
+  base::WideToUTF8(localized_country_name,
+    country_name_length,
+    &country);
+
+  #elif defined(OS_MACOSX)
+  // for more info, see:
+  // https://developer.apple.com/documentation/corefoundation/1543547-cflocalegetvalue?language=objc
+  // https://developer.apple.com/documentation/corefoundation/cflocalekey?language=objc
+  CFLocaleRef cflocale = CFLocaleCopyCurrent();
+  CFStringRef country_code = (CFStringRef)CFLocaleGetValue(
+    cflocale, kCFLocaleCountryCode);
+  CFIndex kCountryNameLength = CFStringGetLength(country_code) + 1;
+  char localized_country_name[kCountryNameLength];
+  if (CFStringGetCString(country_code, localized_country_name,
+    kCountryNameLength, kCFStringEncodingUTF8)) {
+    country = std::string(localized_country_name);
+  }
+
+  #elif defined(OS_LINUX)
+  // for more info, see:
+  // https://en.cppreference.com/w/cpp/locale/setlocale
+  setlocale(LC_ALL, "");
+  country = std::string(setlocale(LC_ALL, NULL));
```

Comment:
you have to check the return value of setlocale, if it returns null and then you pass it to std::string. program crashes.